### PR TITLE
Rename create to module_enabled and add module_depends_on

### PR DESCRIPTION
**Pull request by soerenmartius**

### CODE REVIEW

**Comment by mariux** [comment]
@soerenmartius unit tests do not seem to run?

File: nat.tf [hunk 1]
```diff
@@ -64,15 +64,18 @@ resource "aws_eip" "nat" {
     var.tags
   )
 
-  depends_on = [aws_internet_gateway.internet_gateway]
+  depends_on = [
+    aws_internet_gateway.internet_gateway,
+    var.module_depends_on
```
Comment (mariux) [inline nat.tf:69]:
```suggestion
    var.module_depends_on,
```

File: nat.tf [hunk 2]
```diff
@@ -85,5 +88,8 @@ resource "aws_nat_gateway" "nat" {
 
   # It's recommended to denote that the NAT Gateway depends on the Internet Gateway for the VPC in which the NAT
   # Gateway's subnet is located. https://www.terraform.io/docs/providers/aws/r/nat_gateway.html
-  depends_on = [aws_internet_gateway.internet_gateway]
+  depends_on = [
+    aws_internet_gateway.internet_gateway,
+    var.module_depends_on
```
Comment (mariux) [inline nat.tf:93]:
```suggestion
    var.module_depends_on,
```

File: private-subnets.tf [hunk 3]
```diff
@@ -72,8 +76,10 @@ resource "aws_route" "private_nat" {
   depends_on = [
     aws_internet_gateway.internet_gateway,
     aws_route_table.private,
+    var.module_depends_on
```
Comment (mariux) [inline private-subnets.tf:79]:
```suggestion
    var.module_depends_on,
```

File: nat.tf (hunk 1 above)
Comment (soerenmartius) [reply nat.tf:69]: Why is this important?

Comment (soerenmartius) [reply nat.tf:69]: If this is the enforced style then we should add it to the linters ( `terraform fmt` and `tflint` aren't complaining about it )

Comment (soerenmartius) [reply nat.tf:69]: > lgtm but would love to have a comma in the end when doing multi-line lists...

Can we add this to `terraform fmt` or `tflint` somehow?

**Comment by soerenmartius** [comment]
> @soerenmartius unit tests do not seem to run?

Yeah, we configured the tests not to run on every check-in but only on commits to master or manual promotion ( see https://github.com/mineiros-io/terraform-aws-vpc/blob/master/.semaphore/semaphore.yml#L23 for details ).

This is indeed the desired behaviour. Running the tests takes a while. In case of rather complex modules in which the deployments can take a while and cause costs, we should run the tests manually before merging back to master instead of running them on every commit. Or WDYT?

**Comment by mariux** [comment]
hmmm need to think about this... bad situation as tests should not be green when not run.

File: nat.tf (hunk 1 above)
Comment (soerenmartius) [reply nat.tf:69]: @mariux saw this?

**Comment by soerenmartius** [comment]
> hmmm need to think about this... bad situation as tests should not be green when not run.

Okay, I am gonna create a ticket for this in the morning.

File: private-subnets.tf (hunk 3 above)
Comment (soerenmartius) [reply private-subnets.tf]: resolved in fd7f9a2

File: nat.tf (hunk 2 above)
Comment (soerenmartius) [reply nat.tf]: resolved in fd7f9a2

File: nat.tf (hunk 1 above)
Comment (soerenmartius) [reply nat.tf]: resolved in fd7f9a2

**Pull request merged**